Allow building with macOS 10.14 SDK (will still warn about
"Instance method '-charactersByApplyingModifiers' unknown")

Upstream-Status: Inappropriate [violates DRY]

--- src/gui/platform/darwin/qapplekeymapper.mm.orig
+++ src/gui/platform/darwin/qapplekeymapper.mm
@@ -528,7 +528,7 @@
                 // we first run the event through the Carbon APIs and then
                 // compare the results to Cocoa.
                 auto cocoaModifiers = toCocoaModifiers(qtModifiers);
-                auto *charactersWithModifiers = [NSApp.currentEvent charactersByApplyingModifiers:cocoaModifiers];
+                NSString *charactersWithModifiers = [NSApp.currentEvent charactersByApplyingModifiers:cocoaModifiers];
 
                 QChar cocoaUnicodeKey;
                 if (charactersWithModifiers.length > 0)
